typing/types.mli

```diff
@@ -370,14 +370,15 @@ type type_declaration =
     type_loc: Location.t;
     type_attributes: Parsetree.attributes;
     type_immediate: Type_immediacy.t;
-    type_unboxed: unboxed_status;
+    type_unboxed_default: bool;
+    (* true if the unboxed-ness of this type was chosen by a compiler flag *)
```

Comment:
Two remarks here:
- do you remember what this information is used for?
- another (nicer?) approach would be to store this information as a parameter of `Record_unboxed` and `Variant_unboxed`, for example

```ocaml
type unboxed_setting = On_decl | Global_flag
```

Reply:
It is used in exactly one place, in the implementation of Warning 61:
```
# type t = Foo of int;;
type t = Foo of int
# external blah : int -> t = "blah";;
Warning 61: This primitive declaration uses type t, whose representation
may be either boxed or unboxed. Without an annotation to indicate
which representation is intended, the boxed representation has been
selected by default. This default choice may change in future
versions of the compiler, breaking the primitive implementation.
You should explicitly annotate the declaration of t
with [@@boxed] or [@@unboxed], so that its external interface
remains stable in the future.
external blah : int -> t = "blah"
```
It's a sufficiently obscure and special-case situation that I'd rather not clutter up the frequently used representation information with it.